src/intersection/compute_intersection.cc

```diff
@@ -0,0 +1,1139 @@
+     * (u3  -v3)     = (c3-a3)
+     * 
+     * these are 3 equations for variables s,t
+     * see (4.3) on pg. 19 in DP VF
+     * 
+     * We will solve this using Crammer's rule for the maximal subdeterminant det_ij of the matrix.
+     * s = detX_ij / det_ij
+     * t = detY_ij / det_ij
+     */
+    
+    // starting point of abscissa
+    arma::vec3 A = (*abscissa_)[0].point_coordinates();
+    // direction vector of abscissa
+    arma::vec3 U = plucker_coordinates_abscissa_->get_u_vector();
+    // vertex of triangle side
+    arma::vec3 C = (*triangle_)[side][side%2].point_coordinates();
+    // direction vector of triangle side
+    arma::vec3 V = plucker_coordinates_triangle_[side]->get_u_vector();
+    // right hand side
+    arma::vec3 K = C - A;
+    // subdeterminants det_ij of the system equal minus normal vector to common plane of U and V
+    // det_12 =  (-UxV)[1]
+    // det_13 = -(-UxV)[2]
+    // det_23 =  (-UxV)[3]
+    arma::vec3 Det = -arma::cross(U,V);
+    
+    
+//     A.print();
+//     U.print();
+//     C.print();
+//     V.print();
+//     K.print();
+//     Det.print();
+//     cout <<endl;
+
+    unsigned int max_index = 0;
+    double maximum = fabs(Det[0]);
+    if(fabs((double)Det[1]) > maximum){
+        maximum = fabs(Det[1]);
+        max_index = 1;
+    }
+    if(fabs((double)Det[2]) > maximum){
+        maximum = fabs(Det[2]);
+        max_index = 2;
+    }
+//     DBGMSG("det max: %e\n", maximum);
+    //abscissa is parallel to triangle side
+    if(std::abs(maximum) <= std::sqrt(rounding_epsilon)) return false;
+
+    // map maximum index in {-UxV} to i,j of subdeterminants
+    //              i j
+    // max_index 0: 1 2
+    //           1: 2 0  (switch  due to sign change)
+    //           2: 0 1
+    unsigned int i = (max_index+1)%3,
+                 j = (max_index+2)%3;
+                 
+    double DetX = -K[i]*V[j] + K[j]*V[i];
+    double DetY = -K[i]*U[j] + K[j]*U[i];
+
+    double s = DetX/Det[max_index];   //parameter on abscissa
+    double t = DetY/Det[max_index];   //parameter on triangle side
+
+    // change sign according to side orientation
+    if(RefElement<2>::normal_orientation(side)) t=-t;
+
+//     DBGMSG("s = %f; t = %f\n",s,t);
+
+    // if IP is inside of triangle side
+    if(t >= -geometry_epsilon && t <= 1+geometry_epsilon){
+
+        IP.set_orientation(2);  // set orientation as a pathologic case ( > 1)
+        // possibly set abscissa vertex {0,1}
+        if( fabs(s) <= geometry_epsilon)       { s = 0; IP.set_topology_A(0,0);}
+        else if(fabs(1-s) <= geometry_epsilon) { s = 1; IP.set_topology_A(1,0);}
+        else                         {        IP.set_topology_A(0,1);}   // no vertex, line 0, dim = 1
+        
+        // possibly set triangle vertex {0,1,2}
+        if( fabs(t) <= geometry_epsilon)       { t = 0; IP.set_topology_B(RefElement<2>::interact(Interaction<0,1>(side))[RefElement<2>::normal_orientation(side)],0);}
```

Comment:
Do we even need to detect boundary of the line segment here? It seems useless in auxiliary 1d-2d intersection.